### DISCUSSION TITLE
Classifier task area: show a Next button instead of Done button for required questions

### DISCUSSION THREAD

**Discussion by eatyourgreens** [zooniverse/front-end-monorepo#6629] [Feature Requests]
## Package
lib-classifier 
## Is your feature request related to a problem? Please describe.
When a question is required, but I haven’t answered yet, the classifier shows a disabled Done button. 

## Describe the solution you'd like
If the current step doesn’t end the workflow, it would better to show a disabled Next button, as a visual cue that I haven’t finished the workflow yet. 

https://fe-project.zooniverse.org/projects/aschig/test-premature-done/classify/workflow/19442
![Here, both answers a and b take me to a new step, but the task shows a Done button.](https://user-images.githubusercontent.com/59547/130929983-32de9974-0167-493c-a60b-db0d15d7c196.jpeg)

**Comment by eatyourgreens** [comment]
We don’t show ‘Done & Talk’ on these steps, so there’s already logic in place which says that this isn’t the last step.

**Comment by eatyourgreens** [comment]
On the other hand, if this question wasn’t required, we should be able to prematurely end the workflow here. In that case, there should be an option to go to Talk without selecting an answer.

I’m not sure if an optional question, like that, is good workflow design.

**Comment by srallen** [comment]
I believe we default to show done for two reason:

- Branching single choice could either branch to submit the classification or go on to the next task and we don't know which yet
- I think there was concern with biasing a volunteer to try to select the option that would let them continue? This is vague memory from the early days of PFE, so I'd have to dig for that.

**Comment by eatyourgreens** [comment]
True, but in the branching case we show Done & Talk and Done under the question task eg. the first question for I Fancy Cats. Here, we only show Done, so there must be a next step. Maybe there's a bug in the Done & Talk logic?

**Comment by srallen** [comment]
@eatyourgreens definitely possible there's a bug. What does PFE do here?

**Comment by eatyourgreens** [comment]
PFE also shows Done, but not Done & Talk.
https://www.zooniverse.org/projects/aschig/test-premature-done/classify

If you try the 'no summaries' workflow here, then it does show a disabled Done & Talk button before you've selected an answer.
https://master.pfe-preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/classify

**Comment by eatyourgreens** [comment]
I'm thinking that maybe, if there's no Done & Talk button, then the other disabled button should be 'Next', but I'm not 100% sure.

**Comment by srallen** [comment]
The PFE behavior seems buggy as well to me. It seems that Done & Talk should always be visible before we know what the branch choice is since we've permanently disabled summaries?

**Comment by eatyourgreens** [comment]
The PFE behaviour is a little harder to debug because there's also a case where a project has not enabled Talk (I Fancy Cats is an example of this) so 'Done & Talk' doesn't show, on PFE, because there is no Talk for that project.

It does seem like it shouldn't be possible to see Done without also seeing Done & Talk, in the new classifier at least.

**Comment by goplayoutside3** [comment]
@seanmiller26 I converted this legacy Issue to a Discussion with the label "feature request" for consideration. As far as I can tell, there's no change needed here to match parity with PFE, and more questions than answers about why to show or not show Next when a task is required.
